src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -1012,6 +964,49 @@ bool QgsOracleProvider::determinePrimaryKey()
   return mValid;
 }
 
+void QgsOracleProvider::determinePrimaryKeyFromUriKeyColumn()
+{
+  QString primaryKey = mUri.keyColumn();
+  mPrimaryKeyType = PktUnknown;
+
+  if ( !primaryKey.isEmpty() )
+  {
+    int idx = fieldNameIndex( primaryKey );
+
+    if ( idx >= 0 )
+    {
+      QgsField fld = mAttributeFields.at( idx );
+
+      if ( mUseEstimatedMetadata || uniqueData( mQuery, primaryKey ) )
+      {
+        if ( fld.type() == QVariant::Int ||
+             fld.type() == QVariant::LongLong ||
+             ( fld.type() == QVariant::Double && fld.precision() == 0 ) )
+        {
+          mPrimaryKeyType = PktInt;
+        }
+        else
+        {
+          mPrimaryKeyType = PktFidMap;
+        }
+        mPrimaryKeyAttrs << idx;
+      }
+      else
+      {
+        QgsMessageLog::logMessage( tr( "Primary key field '%1' for view/query not unique." ).arg( primaryKey ), tr( "Oracle" ) );
```

Comment:
emit raiseError instead of this?

Reply:
I have just move the code and it is the same in postgres provider.

Reply:
Ok, let's leave as is then :+1: